core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.controller.spec.ts

```diff
@@ -105,13 +105,69 @@ describe('Translation Modal Controller', function() {
       expect($scope.loadingData).toBe(false);
     });
 
+  it('should not register Contributor Dashboard submit suggestion event when' +
+    ' all images are not copied when suggesting translated text',
+  function() {
+    $httpBackend.flush();
+    spyOn(
+      SiteAnalyticsService,
+      'registerContributorDashboardSubmitSuggestionEvent');
+    $scope.textToTranslate = '<oppia-noninteractive-image alt-with-value=' +
+      '"&amp;quot;Image description&amp;quot;" caption-with-value=' +
+      '"&amp;quot;Image caption&amp;quot;" filepath-with-value="&amp;quot;' +
+      'img_20210129_210552_zbv0mdty94_height_54_width_490.png&amp;quot;">' +
+      '</oppia-noninteractive-image>';
+    $scope.activeWrittenTranslation.html = '';
+    $scope.suggestTranslatedText();
+    expect(
+      SiteAnalyticsService.registerContributorDashboardSubmitSuggestionEvent)
+      .not.toHaveBeenCalledWith('Translation');
```

Comment:
When the test is non-trivial, prefer to add new lines to emphasize test structure, i.e. setup, action, and assertion: https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests. Test structure applies for all unit tests.

```suggestion
    $httpBackend.flush();
    spyOn(
      SiteAnalyticsService,
      'registerContributorDashboardSubmitSuggestionEvent');
    $scope.textToTranslate = '<oppia-noninteractive-image alt-with-value=' +
      '"&amp;quot;Image description&amp;quot;" caption-with-value=' +
      '"&amp;quot;Image caption&amp;quot;" filepath-with-value="&amp;quot;' +
      'img_20210129_210552_zbv0mdty94_height_54_width_490.png&amp;quot;">' +
      '</oppia-noninteractive-image>';
    $scope.activeWrittenTranslation.html = '';
    
    $scope.suggestTranslatedText();
    
    expect(
      SiteAnalyticsService.registerContributorDashboardSubmitSuggestionEvent)
      .not.toHaveBeenCalledWith('Translation');
```

Reply:
Done